### PR TITLE
Adding Set and OrderedDict support in syft.lib.python

### PR DESCRIPTION
**Pull request by tudorcebere**
## Description
Added support for `Set` in syft.lib.python.

Solves #4778 and it's a part of #4515 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by madhavajay** [comment]
@tudorcebere Awesome work!! I tested it by changing all the state_dict to OrderedDict and removed the previous hack and it all works.

Can you merge this PR in?: https://github.com/OpenMined/PySyft/tree/madhava/fix-4836

File: src/syft/lib/python/__init__.py [hunk 1]
```diff
@@ -426,6 +430,41 @@ def create_python_ast() -> Globals:
...
+        ("syft.lib.python.Set.__le__", "syft.lib.python.Bool", Bool),
+        ("syft.lib.python.Set.__len__", "syft.lib.python.Int", Int),
+        ("syft.lib.python.Set.__lt__", "syft.lib.python.Bool", Bool),
+        ("syft.lib.python.Set.__ne__", "syft.lib.python.Bool", Bool),
+        ("syft.lib.python.Set.__or__", "syft.lib.python.Set", Set),
+        ("syft.lib.python.Set.__sub__", "syft.lib.python.Set", Set),
+        ("syft.lib.python.Set.__xor__", "syft.lib.python.Set", Set),
+        ("syft.lib.python.Set.add", "syft.lib.python._SyNone", _SyNone),
+        ("syft.lib.python.Set.clear", "syft.lib.python._SyNone", _SyNone),
+        ("syft.lib.python.Set.difference", "syft.lib.python.Set", Set),
```
Comment (madhavajay) [inline src/syft/lib/python/__init__.py:451]: Could sets be used on lists to reveal particularly private information?

File: src/syft/lib/python/collections/ordered_dict.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    def __len__(self) -> SyPrimitiveRet:
+        res = super().__len__()
+        return PrimitiveFactory.generate_primitive(value=res)
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __ne__(self, other: Any) -> SyPrimitiveRet:
+        res = super().__ne__(other)
+        return PrimitiveFactory.generate_primitive(value=res)
+
+    # @syft_decorator(typechecking=True, prohibit_args=False)
```
Comment (madhavajay) [inline src/syft/lib/python/collections/ordered_dict.py:77]: Any reason why this is commented out?

File: tests/syft/lib/python/collections/ordered_dict/ordered_dict_serde_test.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    dict2._id = UID()
+    assert type(getattr(dict2, "id", None)) is UID
+
+    d3 = OrderedDict({"t1": 1, "t2": 2})
+    dict3 = OrderedDict(d3)
+    assert type(getattr(dict3, "id", None)) is UID
+
+    assert dict1.keys() == dict2.keys()
+    assert dict1.keys() == dict3.keys()
+
```
Comment (madhavajay) [inline tests/syft/lib/python/collections/ordered_dict/ordered_dict_serde_test.py:34]: Should we add a quick test to make sure our OrderedDict always respects the insertion order?

Comment (madhavajay) [reply tests/syft/lib/python/collections/ordered_dict/ordered_dict_serde_test.py:34]: I resolved this with a commit.

File: src/syft/lib/python/collections/ordered_dict.py (hunk 2 above)
Comment (madhavajay) [reply src/syft/lib/python/collections/ordered_dict.py]: I resolved this with a commit.

**Review by madhavajay — APPROVED** [review]
LGTM!
